lib/spoom/coverage.rb

```diff
@@ -11,10 +11,19 @@ module Spoom
   module Coverage
     extend T::Sig
 
-    sig { params(path: String).returns(Snapshot) }
-    def self.snapshot(path: '.')
+    sig { params(path: String, rbi: T::Boolean).returns(Snapshot) }
+    def self.snapshot(path: '.', rbi: false)
+      config = sorbet_config(path: path)
+      config.allowed_extensions.push(".rb", ".rbi") if config.allowed_extensions.empty?
+
+      new_config = Sorbet::Config.new
+      new_config.paths.concat(config.paths)
+      new_config.ignore.concat(config.ignore)
+      new_config.allowed_extensions.concat(config.allowed_extensions.reject { |ext| !rbi && ext == ".rbi" })
```

Comment:
It might be a bit hacky but a quick and dirty way of deep cloning `Config` in an encapsulated way would be to add the method:
```
def deep_clone
  Marshal.load(Marshal.dump(self))
end
```
to the `Config` class.

With that, this code becomes:
```suggestion
      new_config = config.deep_clone
      new_config.allowed_extensions.concat(config.allowed_extensions.reject { |ext| !rbi && ext == ".rbi" })
```

Reply:
This looks less good when you add all the casts to make it statically valid.

I factorized away the copy of the object in 005dc93.